pkg/api/v20170701/types.go

```diff
@@ -98,15 +112,15 @@ const (
 
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
-	OrchestratorType    OrchestratorType    `json:"orchestratorType"`
+	OrchestratorType    OrchestratorType    `json:"orchestratorType" validate:"required"`
 	OrchestratorVersion OrchestratorVersion `json:"orchestratorVersion"`
 }
 
 // MasterProfile represents the definition of master cluster
 type MasterProfile struct {
-	Count                    int    `json:"count"`
-	DNSPrefix                string `json:"dnsPrefix"`
-	VMSize                   string `json:"vmSize"`
+	Count                    int    `json:"count" validate:"required"`
```

Comment:
Master Count has a default value in the implementation of `az cli`. 
You can think of `az cli` is the helper for user to put together the api model. This is the validation tag for this api model.

It works, after `az cli` generate the api model. We validate if it provide the required fields. 
Keep in mind, this validate tag only works on validate the generated api model.